types/instance.go

```diff
@@ -70,3 +80,75 @@ func (i Instance) GetItem() interface{} {
 func (i Instance) GetType() string {
 	return "instance"
 }
+
+type JsonResponseBody struct {
+	Json map[string]interface{}
+}
+
+type RemoteResponse struct {
+	Code  int
+	Body  string
+	Json  JsonResponseBody
+	Error bool
+}
+
+// GetUrl returns the result of an HTTP request to the instance
+func (i Instance) GetUrl(path string, port string) RemoteResponse {
+	var uri string
+	checkPort := "80"
+
+	if port == "" {
+		uri = fmt.Sprintf("http://%s%s", i.IpAddress, path)
+	} else {
+		uri = fmt.Sprintf("http://%s:%s%s", i.IpAddress, port, path)
+		checkPort = port
+	}
+
+	if tcp_port_test(i.IpAddress, checkPort) == false {
+		return RemoteResponse{999, "", JsonResponseBody{}, true}
+	}
+
+	log.Debugf("Making HTTP request to %s", uri)
+	resp, err := http.Get(uri)
```

Comment:
Default timeout is unlimited in Go. could you please set a timeout for this call?